walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -72,40 +94,36 @@ private boolean eventParticipantInMeeting(
 
       // Only move reference points of end of last event if the cur event end point is later
       // than the end of the event with the latest end point so far.
-      endOfEarlierEvent = max(endOfEarlierEvent, curEvent.getWhen().end());
+      endOfEarlierEvent = Math.max(endOfEarlierEvent, curEvent.getWhen().end());
     }
 
     return openMeetingTimes;
   }
 
   /**
-   * Returns a collection of all meeting times that will work for all attendees' schedules.
+   * Returns a collection of all meeting times that will work for attendees' schedules.
    *
-   * <p>To determine the meeting times ({@code TimeRange}s) that works for all attendees, the event
-   * list is filtered and sorted so that {@code getMeetingTimes()} can determine all feasible
-   * meeting times. More specifically, filtered means that an event is removed if it does not
-   * contain at least one participant that is included in the meeting request. Both filtering and
-   * then sorting are accomplished using a stream of Event objects.
+   * <p>Initially, meeting times ({@code TimeRange}s) that will works for both mandatory and
+   * optional attendees are identified. If there are no meeting times that work for all attendees,
+   * only meeting times with mandatory attendees are identified and returned.
    *
    * @param events The Collection of events that are used to determine what periods of time that the
    *     meeting can take place.
    * @param request The Meeting request that contains the requirements for potential meetings.
    * @return A Collection of the feasible meeting {@code TimeRange}s.
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    List<Event> eventList =
-        events.stream()
-            .filter(event
-                -> eventParticipantInMeeting(
-                    new HashSet<String>(event.getAttendees()), request.getAttendees()))
-            .sorted(new Comparator<Event>() {
-              @Override
-              public int compare(Event a, Event b) {
-                return Long.compare(a.getWhen().start(), b.getWhen().start());
-              }
-            })
-            .collect(Collectors.toList());
+    Collection<TimeRange> mandatoryAndOptionalTimesList =
+        getMeetingTimes(events, request, /*considerOptionalAttendees=*/true);
+
+    // In the special case where there are no mandatory attendees, at least one optional attendee,
+    // and no times work out, return an empty list rather than considering only mandatory attendees.
+    // This will result in a time range for the whole day and is not desirable.
+    if (!mandatoryAndOptionalTimesList.isEmpty()
+        || request.getAttendees().isEmpty() && !request.getOptionalAttendees().isEmpty()) {
```

Comment:
This condition is hard to read since the order of || vs && comes into play. It would be better if you could add parentheses around a group of 2 conditions to make it more readable. Example: (A || B) && C vs A || (B && C).

Reply:
Yeah that is more readable that way. Fixed.

I think the reason I omitted it back then was because I just took digital system design last semester and we did a LOOT of boolean algebra so I got used to typing no parentheses when needed for speed in that class, haha.